### DISCUSSION TITLE
Logging multiple scalars to a single wandb chart

### DISCUSSION THREAD

**Discussion by hogru** [Lightning-AI/pytorch-lightning#11960] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
During `fit()` I want to log multiple scalars to a _single_ chart using the wandb logger. I am trying to achieve this with this example code:

`self.log('val/top-k', {'1': 0.6, '5': 0.8, '10': 0.9})`

On wandb this creates _three_ charts with the headings `val/top-k.1` and `(...).5` and `(...).10` resp.

Am I doing this wrong or do I need to call the wandb logger directly (which I try to avoid to keep the code logger agnostic).

Thanks!

**Comment by rohitgr7** [comment, marked as answer]
hey @hogru!

can you share some references to how wandb plots multiple charts? I don't think there is a direct API. Personally, I used to log them individually and combine them in a single chart afterward. But if there is an option to do that via its API, then we can look more and think about adding support for it if it's not already there.

**Comment by hogru** [comment]
Hello @rohitgr7,

thank you for the quick response! You are right. I am new to both Pytorch Lightning and W&B and based on TensorBoard experience and looking at the possibility to log a dict I was assuming that there "must" be a way to do this. Now that I have seen that W&B allows to customize the graphs in their UI (and more importantly it keeps the setup across runs which is nice) I can do that within the W&B Web UI.

Thanks again!
